p2p/enode/nodedb.go

```diff
@@ -262,7 +261,7 @@ func (db *DB) storeInt64(key []byte, n int64) error {
 	blob := make([]byte, binary.MaxVarintLen64)
 	blob = blob[:binary.PutVarint(blob, n)]
 	return db.kv.Update(db.ctx, func(tx kv.RwTx) error {
-		return tx.Put(kv.Inodes, libcommon.CopyBytes(key), blob)
+		return tx.Put(kv.Inodes, key, blob)
```

Comment:
Is there no possible race here, to have removed `CopyBytes` and instead directly using the slice?

Reply:
`db.Update` (and `db.View` and `db.Batch`) - doesn't create goroutine and doesn't move key/value to another goroutine. It means `race` can't happen here. 

You probably did mean - not `race` but `does tx.Put copy inside? is it safe to modify `key` variable after tx.Put?`. Yes - `tx.Put` does copy key and value into MMAP (or in-case of in-memory-mutation - it also does copy before put to `map[string]string`). 

We have not so much places which does internal copy when write: 
- ETL (method .Collect)
- DB  (method .Put)

About `kv.RoTx` zero-copy reads: 
- `tx.GetOne` does return pointer to MMAP'ed region - which valid until `tx.Rollback`. Can don't copy key/value if don't need them outside of current `RoTx`. 
- `tx.Cursor` - same as `tx.GetOne`
- `tx.Range` (iterators) - have lower guarantees - key/value are valid until next call of `.Next`.